lib/livebook/intellisense/identifier_matcher.ex

```diff
@@ -176,6 +176,57 @@ defmodule Livebook.Intellisense.IdentifierMatcher do
     end
   end
 
+  @doc """
+
+  """
+  @spec fetch_identifier(
+          charlist(),
+          {:module, module()} | {:function | :type, name :: atom(), arity :: pos_integer()}
+        ) ::
+          {:ok, cell_id :: String.t(), line :: pos_integer()} | :error
+  def fetch_identifier(path, identifier)
+
+  def fetch_identifier(path, {:module, module}) do
+    with {:ok, {:raw_abstract_v1, annotations}} <- beam_lib_chunks(path, :abstract_code) do
+      {:attribute, anno, :module, ^module} =
+        Enum.find(annotations, &match?({:attribute, _, :module, _}, &1))
+
+      {:ok, cell_id_from_module(path), :erl_anno.line(anno)}
+    end
+  end
+
+  def fetch_identifier(path, {:function, name, arity}) do
+    with {:ok, {:debug_info_v1, _, {_, metadata, _}}} <- beam_lib_chunks(path, :debug_info) do
```

Comment:
Let's also match on the metadata version:

```suggestion
    with {:ok, {:debug_info_v1, _, {:elixir_v1, metadata, _}}} <- beam_lib_chunks(path, :debug_info) do
```

Reply:
Btw, this logic only works for Elixir. For Erlang, you need to do something similar to types, but we can worry about it later.